yt/visualization/plot_modifications.py

```diff
@@ -683,16 +686,49 @@ def __call__(self, plot):
                 linewidth=self.linewidth)
             plot._axes.add_collection(grid_collection)
 
-            if self.draw_ids:
-                visible_ids = np.logical_and(
+            visible_ids = np.logical_and(
                     np.logical_and(xwidth > self.min_pix_ids,
                                    ywidth > self.min_pix_ids),
                     np.logical_and(levels >= min_level, levels <= max_level))
-                for i in np.where(visible_ids)[0]:
-                    plot._axes.text(
-                        left_edge_x[i] + (2 * (xx1 - xx0) / xpix),
-                        left_edge_y[i] + (2 * (yy1 - yy0) / ypix),
-                        "%d" % block_ids[i], clip_on=True)
+
+            if self.id_loc and not self.draw_ids:
+                warnings.warn("Supplied id_loc but draw_ids is False. Not drawing grid ids")
+
+            if self.draw_ids:
+                id_loc = self.id_loc.lower()
+                if id_loc == "lower left":
+                    for i in np.where(visible_ids)[0]:
+                        plot._axes.text(
+                            left_edge_x[i] + (2 * (xx1 - xx0) / xpix),
+                            left_edge_y[i] + (2 * (yy1 - yy0) / ypix),
+                            "%d" % block_ids[i], clip_on=True)
+                elif id_loc == "lower right":
+                    for i in np.where(visible_ids)[0]:
+                        plot._axes.text(
+                            right_edge_x[i] - ((10*len(str(block_ids[i]))-2) \
+                                                * (xx1 - xx0) / xpix),
+                            left_edge_y[i] + (2 * (yy1 - yy0) / ypix),
+                            "%d" % block_ids[i], clip_on=True)
+                elif id_loc == "upper left":
+                    for i in np.where(visible_ids)[0]:
+                        plot._axes.text(
+                            left_edge_x[i] + (2 * (xx1 - xx0) / xpix),
+                            right_edge_y[i] - (12 * (yy1 - yy0) / ypix),
+                            "%d" % block_ids[i], clip_on=True)
+                elif id_loc == "upper right":
+                    for i in np.where(visible_ids)[0]:
+                        plot._axes.text(
+                            right_edge_x[i] - ((10*len(str(block_ids[i]))-2) \
+                                                * (xx1 - xx0) / xpix),
+                            right_edge_y[i] - (12 * (yy1 - yy0) / ypix),
+                            "%d" % block_ids[i], clip_on=True)
+                else:
```

Comment:
I'd raise a RuntimeError instead of warning - some people have yt's logging turned off and won't see the message

Reply:
Raise RuntimeError if the location is bad? (I can't see all of the region)
Is there a yt-specific way to raise the error?